### ISSUE TITLE
ADE20k results

### ISSUE THREAD

**Issue by prinshul** [LiheYoung/UniMatch-V2#17]
I am trying to recreate the results from the paper UniMatch-V2. First of all, excellent work on the paper, and it was great reading it. I can reproduce the results for UniMatch-V2 on Pascal, and Coco, but unable able to do the same for ADE20k. I have used the scripts in your code to generate the results on the validation set as provided by you. For Pascal, and Coco I was able to generate the same results as in the paper precisely, but for ADE20k, using the trained models provided in this [link](https://huggingface.co/LiheYoung/UniMatch-V2/tree/main), I was not getting the results as expected. I believe I have missed some pre-processing steps for ADE20k which you guys have used or maybe the trained models uploaded are for a different experiment? Can you please provide any additional information regarding this? I have used the evaluate function provided by you in the script supervised.py. The only preprocessing used during validation (as per my observation) for all the dataset was normalization. But clearly, something is missing, cause I am extremely off the results in the paper for ADE20k. For example, UniMatch-V2 using DinoV2-Base for 1/16 labels, I got 0.06458 mIOU using this [model ](https://huggingface.co/LiheYoung/UniMatch-V2/blob/main/ade20k_unimatch_v2_1_16labels_dinov2_base.pth), while the expected mIOU was 46.7. 
Please let me know what your suggestions are regarding this. Any input from your end will be extremely helpful in my research.

**Comment by LiheYoung** [comment]
Did you download our processed ADE20K masks? https://github.com/LiheYoung/UniMatch-V2#datasets

**Comment by prabal09** [comment]
Yes!! I have downloaded the processed ADE20K masks as provided by you in above link. I have also used these [samples](https://github.com/LiheYoung/UniMatch-V2/blob/main/splits/ade20k/val.txt) to obtain the validation mIOU.

**Comment by prinshul** [comment]
Hi @LiheYoung can you please verify the Ade20k models on hugging face or give us the code/instructions to reproduce Ade20k results?

Thanks.

**Comment by prinshul** [comment]
Hi @LiheYoung we have verified all models. Except ADE20K all others are reproducing with the results given in the paper. So can you suggest what would be going wrong in ADE20K.

**Comment by LiheYoung** [comment]
I am a little busy this week. Please wait for another several days. Could I
know what this ADE20K ckp is used for? Because you may train your own
models using our code.

PRASHANT PANDEY ***@***.***>于2025年2月19日 周三18:15写道：

**Comment by prinshul** [comment]
Sure. I think we can do that. We will give it a try. Thanks.
